demos/text_detection_demo/cpp/include/cnn.hpp

```diff
@@ -39,3 +45,44 @@ class Cnn {
     double time_elapsed_;
     size_t ncalls_;
 };
+
+class EncoderDecoderCNN : public Cnn {
+  public:
+
+    void Init(const std::string &model_path, Core & ie, const std::string & deviceName,
+              const cv::Size &new_input_resolution = cv::Size());
+
+    InferenceEngine::BlobMap Infer(const cv::Mat &frame);
+    void setInOutNames(const std::string out_enc_hidden_name,
+                      const std::string out_dec_hidden_name,
+                      const std::string in_dec_hidden_name,
+                      const std::string features_name,
+                      const std::string in_dec_symbol_name,
+                      const std::string out_dec_symbol_name,
+                      const std::string logits_name);
+  private:
+    InferRequest infer_request_encoder_;
+    InferRequest infer_request_decoder_;
+    std::vector<std::string> input_names_decoder;
+    std::vector<std::string> output_names_encoder;
+    std::vector<std::string> output_names_decoder;
+    std::string features_name_;
+    std::string out_enc_hidden_name_;
+    std::string out_dec_hidden_name_;
+    std::string in_dec_hidden_name_;
+    std::string in_dec_symbol_name_;
+    std::string out_dec_symbol_name_;
+    std::string logits_name_;
+    InputInfo::Ptr input_info_decoder_input;
+};
+
+class CnnFactory {
+public:
+    Cnn* create(unsigned type) {
+        switch (type) {
+            case 0: return new Cnn();
+            case 1: return new EncoderDecoderCNN();
+            default: return new Cnn();
```

Comment:
I cant find corresponding `delete`. Can you use `unique_ptr` instead?

Reply:
Done